execute/plugin_functions.go

```diff
@@ -227,15 +269,47 @@ func truncateObservation(
 	// If the encoded observation is too large, start filtering data.
 	for len(encodedObs) > maxSize {
 		for _, chain := range chains {
-			if len(observation.CommitReports[chain]) > 1 {
-				observation = truncateLastCommit(observation, chain)
-			} else {
+			commits := observation.CommitReports[chain]
+			if len(commits) == 0 {
+				continue
+			}
+			lastCommit := &commits[len(commits)-1]
+			seqNum := lastCommit.SequenceNumberRange.Start()
+
+			observation.PseudoDeleted[chain] = make(map[cciptypes.SeqNum]bool)
+			for seqNum <= lastCommit.SequenceNumberRange.End() {
```

Comment:
You're deleting messages in the commit from first to last, did you mean to delete them last to first?

```suggestion
			seqNum := lastCommit.SequenceNumberRange.End()

			observation.PseudoDeleted[chain] = make(map[cciptypes.SeqNum]bool)
			for seqNum >= lastCommit.SequenceNumberRange.Start() {
```

Reply:
It won't be executed on chain if seqNr is later?